packages/react-pagination/test/setup.ts

```diff
@@ -0,0 +1,7 @@
+import { cleanup } from "@testing-library/react";
+import { afterEach } from "vitest";
+
+// Clears the JSDom after each "it" statement
+afterEach(() => {
+  cleanup();
+});
```

Comment:
I didn't know this was necessary. What happens if you don't do this exactly?

Also, how would you feel about moving this into the `usePagination.test.ts` file for simplicity's sake? I'm used to the setup being a place where global mocks that aren't already provided by jest/vitest could be installed (like if you install a plugin for mocking Date for example). But if you say this is the right way to do it, that's cool of course!

Reply:
I noticed we have the same in `react-essentials`, which contains many test files of course, so I can see why you would do this cleanup in the setup file. Still curious what it does though :)

Reply:
https://testing-library.com/docs/react-testing-library/api/#cleanup
React testing library attaches the cleanup function when global `aftereach` is available. Since `vitest` doesn't have global variables like `jest` you need to manually invoke `cleanup` function.